internal/server/pool_test.go

```diff
@@ -53,7 +54,455 @@ func TestPoolSpike(t *testing.T) {
 	wg.Wait()
 
 	be.Equal(rt, 5, p.Stats().Idle)
-	be.NilErr(rt, p.Close())
+}
+
+func TestMaxOpenConnsUpdate(t *testing.T) {
+	t.Run("basic", func(t *testing.T) {
+		cfg := genBasePoolConfig()
+		cfg.MaxOpen = 3
+		cfg.MaxIdle = 2
+
+		rt := be.Relaxed(t)
+
+		p, err := NewPool(cfg)
+		be.NilErr(rt, err)
+		defer func() { be.NilErr(rt, p.Close()) }()
+
+		conn0, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		conn1, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		conn2, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		be.Equal(t, 3, p.Stats().OpenConnections)
+
+		p.SetMaxOpenConns(2)
+
+		p.ReleaseConn(conn0)
+		p.ReleaseConn(conn1)
+
+		// Ensure that extra connections are not added back.
+		be.Equal(t, 2, p.Stats().OpenConnections)
```

Comment:
Is there any order of connections to be checked? I mean once we set with `SetMaxOpenConns`, do we know which connection would be removed?

Reply:
Not really, doesn't matter.